src/dlr_tensorflow2/dlr_tensorflow2.cc

```diff
@@ -65,58 +65,37 @@ TF_Output Tensorflow2Model::ParseTensorName(const std::string& t_name) {
   return oper_out;
 }
 
-void Tensorflow2Model::DetectInputs() {
-  size_t pos = 0;
-  TF_Operation* op;
-  while ((op = TF_GraphNextOperation(graph_, &pos)) != nullptr) {
-    const std::string op_type = TF_OperationOpType(op);
-    const int n_in = TF_OperationNumInputs(op);
-    const int n_out = TF_OperationNumOutputs(op);
-    const std::string op_name = TF_OperationName(op);
-    if (op_type == "Placeholder" && n_in == 0 && n_out == 1 && ignored_names_.count(op_name) == 0) {
-      input_names_.push_back(op_name + ":0");
+void Tensorflow2Model::DetectInputsAndOutputs(const InputOutputType& inputs,
+                                              const InputOutputType& outputs) {
+  for (auto& el : inputs) {
+    const tensorflow::TensorInfo& ti = el.second;
+    input_names_.push_back(ti.name());
+    const tensorflow::TensorShapeProto& shape = ti.tensor_shape();
```

Comment:
We can save TensorNames for our internal usage. But, it would be convenient for our users if they can use/see human readable input/output names when they use DLR API:
- SetInput(name, ...)
- GetOutputName(index)

Reply:
2. We also need to update test cases in `dlr_tensorflow2_test.cc` ( GetDLRInputName and GetDLROutputName should expect short input/output names)

Reply:
updated.
1. full tensor names are saved in `input_tensor_names_` and `output_tensor_names_`, the DLR C APIs are added.
2. updated the unit tests `dlr_tensorflow2_test.cc`